src/sentry/api/serializers/models/group.py

```diff
@@ -973,6 +978,56 @@ def get_attrs(self, item_list, user):
                     attrs[item].update({"filtered_stats": filtered_stats[item.id]})
                 attrs[item].update({"stats": stats[item.id]})
 
+            if self._expand("sessions"):
+                cache_keys = list(
+                    {self._build_session_cache_key(item.project_id) for item in item_list}
+                )
+                cache_data = cache.get_many(cache_keys)
+                missed_items = []
+                for item, cache_key in zip(item_list, cache_keys):
+                    num_sessions = cache_data.get(cache_key)
+                    if num_sessions is None:
+                        missed_items.append(item)
+                    else:
+                        attrs[item].update(
+                            {
+                                "sessionPercent": self._get_session_percent(
+                                    attrs[item]["times_seen"], num_sessions
+                                )
+                            }
+                        )
+
+                if missed_items:
+                    filters = {"project_id": list({item.project_id for item in missed_items})}
+                    if self.environment_ids:
+                        filters["environment"] = self.environment_ids
+
+                    result_totals = raw_query(
+                        selected_columns=["sessions"],
+                        dataset=Dataset.Sessions,
+                        start=self.start,
+                        end=self.end,
+                        filter_keys=filters,
+                        groupby=["project_id"],
+                        referrer="serializers.GroupSerializerSnuba.session_totals",
+                    )
+                    for data in result_totals["data"]:
+                        cache_key = self._build_session_cache_key(data["project_id"])
+                        cache.set(cache_key, data["sessions"], 3600)
+
+                    for item in missed_items:
+                        attrs[item].update({"sessionPercent": None})
+                        for data in result_totals["data"]:
+                            if data["project_id"] == item.project_id:
+                                attrs[item].update(
+                                    {
+                                        "sessionPercent": self._get_session_percent(
+                                            attrs[item]["times_seen"], data["sessions"]
+                                        )
+                                    }
+                                )
```

Comment:
It might be better to just put the results into a dict in the previous loop, and then iterate through `missed_items` here and add them to `attrs`, that way we avoid the O(n2) looping.